src/component/administration/user/CreateNewUserForm.tsx

```diff
@@ -0,0 +1,285 @@
+import * as React from "react";
+import { useEffect, useRef, useState } from "react";
+import { doesUsernameExists } from "../../../action/AsyncUserActions";
+import { useDispatch } from "react-redux";
+import { ThunkDispatch } from "../../../util/Types";
+import AsyncActionStatus from "../../../action/AsyncActionStatus";
+import { AsyncActionSuccess } from "../../../action/ActionType";
+import * as _ from "lodash";
+import ValidationResult, {
+  Severity,
+} from "../../../model/form/ValidationResult";
+import Utils from "../../../util/Utils";
+import { useI18n } from "../../hook/useI18n";
+import PromiseTrackingMask from "../../misc/PromiseTrackingMask";
+import {
+  Button,
+  Col,
+  Collapse,
+  Container,
+  Form,
+  Row,
+  UncontrolledTooltip,
+} from "reactstrap";
+import EnhancedInput, { LabelDirection } from "../../misc/EnhancedInput";
+import Toggle from "react-bootstrap-toggle";
+import { TOGGLE_STYLE } from "../../term/IncludeImportedTermsToggle";
+import { UserAccountData } from "../../../model/User";
+
+interface CreateNewUserFormProps {
+  onSubmit: (userAccountData: UserAccountData) => void;
+}
+
+const CreateNewUserForm: React.FC<CreateNewUserFormProps> = (props) => {
+  const dispatch: ThunkDispatch = useDispatch();
+  const { i18n } = useI18n();
+  const { onSubmit } = props;
+
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [passwordConfirmation, setPasswordConfirmation] = useState("");
+  const [usernameExists, setUsernameExists] = useState(false);
+  const [emailPassword, setEmailPassword] = useState(true);
+
+  // ref is required to keep the same debounced function between renders
+  const fetchUsernameExists = useRef(
+    _.debounce((username) => {
+      dispatch(doesUsernameExists(username)).then((action) => {
+        if (action.status === AsyncActionStatus.SUCCESS) {
+          const actionSuccess = action as AsyncActionSuccess<boolean>;
+          setUsernameExists(actionSuccess.payload);
+        }
+      });
+    }, 250)
```

Comment:
What is this magical constant? What does it do? If it is necessary here, at least explain its use and value in a comment.